src/runtime_src/core/tools/xbutil2/OO_P2P.cpp

```diff
@@ -349,6 +363,12 @@ OO_P2P::execute(const SubCmdOptions& _options) const
     throw xrt_core::error(std::errc::operation_canceled);
   }
 
-  for (auto& device : deviceCollection)
-    p2p(device.get(), string2action(m_action), false);
+  try {
+    p2p(deviceCollection[0].get(), action, XBU::getForce());
+  } catch (const xrt_core::system_error& ex) {
+    std::cerr << "ERROR: " << ex.what() << std::endl;
+    throw xrt_core::error(std::errc::operation_canceled); 
+  }
+  std::cout << print_success_msg(action) << std::endl;
```

Comment:
Please refactor to something cleaner.  Something like:

```
// Help the user know what their next steps are
switch (action) {
   case action_type::enable:
    std::cout <<  "Please WARM reboot the machine to enable P2P now.\n";
    break;
   
  case action_type::disable:
    std::cout << "Please WARM reboot the machine to disable P2P now.\n";
    break;

  case action_type::validate:
    std::cout << "P2P validated successfully.\n";
}
```
Note: Since we are using modern c++ compilers, if in the future the action_type enumeration should get another value, the compilers will detect that not all switch values are covered and the compilation will stop.